controllers/PlagiarismIthenticateActionHandler.inc.php

```diff
@@ -0,0 +1,385 @@
+		$context = $request->getContext();
+		$user = $request->getUser();
+
+		/** @var SubmissionFileDAO $submissionFileDao */
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO');
+		$submissionFile = $submissionFileDao->getById($args['submissionFileId']); /** @var SubmissionFile $submissionFile */
+		$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /** @var SubmissionDAO $submissionDao */
+		$submission = $submissionDao->getById($submissionFile->getData('submissionId')); /** @var Submission $submission*/
+
+		/** @var \IThenticate $ithenticate */
+		$ithenticate = static::$_plugin->initIthenticate(
+			...static::$_plugin->getServiceAccess($context)
+		);
+
+		if (!static::$_plugin->createNewSubmission($request, $user, $submission, $submissionFile, $ithenticate)) {
+			$this->generateUserNotification(
+				$request,
+				NOTIFICATION_TYPE_ERROR, 
+				__('plugins.generic.plagiarism.action.submitSubmission.error')
+			);
+			return $this->triggerDataChangedEvent($submissionFile);
+		}
+
+		$this->generateUserNotification(
+			$request,
+			NOTIFICATION_TYPE_SUCCESS, 
+			__('plugins.generic.plagiarism.action.submitSubmission.success')
+		);
+
+		return $this->triggerDataChangedEvent($submissionFile);
+	}
+
+	/**
+	 * Accept the EULA, stamp it to proper entity (Submission/User or both) and may run
+	 * one of following intended action
+	 * 	- Upload submission file
+	 * 	- Schedule similarity report generation process
+	 *  - Refresh the similarity report scores
+	 *
+	 * @param array $args
+	 * @param Request $request
+	 */
+	public function acceptEulaAndExecuteIntendedAction($args, $request) {
+		$context = $request->getContext();
+		$user = $request->getUser();
+
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /** @var SubmissionFileDAO $submissionFileDao */
+		$submissionFile = $submissionFileDao->getById($args['submissionFileId']);
+		$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /** @var SubmissionDAO $submissionDao */
+		$submission = $submissionDao->getById($submissionFile->getData('submissionId'));
+
+		$confirmSubmissionEula = $args['confirmSubmissionEula'] ?? false;
+
+		if (!$confirmSubmissionEula) {
+
+			$templateManager = $this->getEulaConfirmationTemplate(
+				$request,
+				$args,
+				$context,
+				$submission,
+				$submissionFile
+			);
+
+			SessionManager::getManager()->getUserSession()->setSessionVar('confirmSubmissionEulaError', true);
+
+			return new JSONMessage(
+				true,
+				$templateManager->fetch(static::$_plugin->getTemplateResource('confirmEula.tpl'))
+			);
+        }
+
+		if (!$submission->getData('ithenticateEulaVersion')) {
+			static::$_plugin->stampEulaToSubmission($context, $submission);
+		}
+
+		if (!$user->getData('ithenticateEulaVersion')) {
+			static::$_plugin->stampEulaToSubmittingUser($context, $submission, $user);
+		}
+
+		$onAcceptAction = $args['onAcceptAction'];
+
+		return $this->{$onAcceptAction}($args, $request);
```

Comment:
This is particularly dangerous! It allows the end user to execute any function on the handler.

Reply:
removed .